core/coreapi/dag.go

```diff
@@ -56,6 +61,244 @@ func (api *dagAPI) Session(ctx context.Context) ipld.NodeGetter {
 	return dag.NewSession(ctx, api.DAGService)
 }
 
+// RootMeta is the metadata for a root pinning response
+type RootMeta struct {
+	Cid         cid.Cid
+	PinErrorMsg string
+}
+
+// ImportMany imports CAR files into the blockstore and returns the root CIDs. If pinning was requested errors that
+// occurred when importing each root will be returned as well.
+//
+// This function's API is not yet stable
+func (api *dagAPI) ImportMany(ctx context.Context, directory files.Directory, doPinRoots bool) (<-chan RootMeta, <-chan error) {
```

Comment:
Really not a fan of this function signature, but it does emulate what we had before. Up for suggestions, or maybe we just merge this PR since it's better than what we've got and work on improving this later. Marked this as unstable to mark my dissatisfaction.

The use of files.Directory seems pretty suspect here too, should it be some typed iterator/slice/channel of ReaderClosers?

Reply:
A simpler `Import` function could also be added here separate from what's used by the commands lib.